### PR TITLE
lint gha job and remove travis

### PR DESCRIPTION
**Pull request by crazy-max**
use golangci-lint and adds ci validation

this was the last piece in travis so we can remove travis config. I also found the `pass` tests were not covered because gpg key and pass was not initialized. Now pass tests run both on ci and in the Dockerfile.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:
 
 env:
   DESTDIR: ./bin
-  GO_VERSION: 1.16.7
+  GO_VERSION: 1.18.5
```
Comment (thaJeztah) [inline .github/workflows/build.yml:14]: Ah, whoops, probably my bad (forgot to update after rebasing)

File: Makefile [hunk 2]
```diff
@@ -50,28 +46,13 @@ test:
...
-vet_linux:
-	go vet ./secretservice
-
 lint:
-	for p in `go list ./... | grep -v /vendor/`; do \
-		golint $$p ; \
-	done
-
-fmt:
-	gofmt -s -l `ls **/*.go | grep -v vendor`
```
Comment (thaJeztah) [inline Makefile:71]: Maybe this one is still handy to have to do a quick `gofmt` locally

File: ci/before_script_linux.sh [hunk 3]
```diff
@@ -1,22 +0,0 @@
-set -ex
```
Comment (thaJeztah) [inline ci/before_script_linux.sh:1]: Do we want to keep the script, in case we need to re-generate the fixtures?

**Review by thaJeztah — COMMENTED** [review]
two comments, but otherwise looks good, thanks!

File: ci/before_script_linux.sh (hunk 3 above)
Comment (crazy-max) [reply ci/before_script_linux.sh:1]: Don't think it's necessary but sure

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Pull request merged**